release.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+#
+# This script contains the logic for build and push Vulcan
+# checks available in `cmd` folder to Docker Hub registry.
+
+# shellcheck disable=SC1091
+
+# Force build and push indepently if conditions are meet
+# Force does not override EXCLUDE_LIST_FILE
+FORCE_BUILD_AND_PUSH="${FORCE_BUILD_AND_PUSH:-false}"
+# Add time metrics to output
+PRINT_METRICS="${PRINT_METRICS:-true}"
+# Checks to exclude from build and push
+EXCLUDE_LIST_FILE="${EXCLUDE_LIST_FILE:-_scripts/exclude.lst}"
+# Sleep 1 second between registry calls
+REGISTRY_RATE_LIMIT="${REGISTRY_RATE_LIMIT:-true}"
+BASE_PATH="$PWD"
+
+# Load Libraries
+. _scripts/libgit.sh
+. _scripts/libdocker.sh
+
+# Load required env vars
+eval "$(git_env)"
+eval "$(dkr_env)"
+
+# Fetch dependency version and modification timestamp
+dep_version=$(git_commit_id go.mod)
+dep_timestamp=$(git_timestamp go.mod)
+
+# Fetch branch and check mode (master or experimental)
+branch=$(git_branch .)
+check_branch=${TRAVIS_BRANCH:-$branch}
+check_mode=${TRAVIS_BRANCH:-$branch}
+if [ "$check_mode" != "master" ]; then
+    check_mode="experimental"
+fi
+
+# Download go dependencies
+go mod download
+
+# Iterate over all checks
+for cf in cmd/*; do
+    ts_start=$(date +"%s")
+    # Verify only directories are processed
+    if [ ! -d "$cf" ]; then
+        continue
+    fi
+    check=$(basename "$cf")
+    # Verify if check is in the exlude list
+    if grep -q "$check" "$EXCLUDE_LIST_FILE"; then
+        echo "Skip build and push for check: [$check] - EXCLUDED"
+        continue
+    fi
+    # Fetch check version and modification timestamp
+    check_version=$(git_commit_id "$cf")
+    check_timestamp=$(git_timestamp "$cf")
+
+    # Check if check has been pushed due to code or dependency update
+    pushed_by_dependency=$(dkr_image_exists "$check" "dep-$dep_version")
+    if [[ $REGISTRY_RATE_LIMIT == true ]]; then sleep 1; fi
+    pushed_by_check_code=$(dkr_image_exists "$check" "$check_version")
```

Comment:
Do we need to take into account the dep version? this dep version shoud be explicit in the repo then in the commit.

Reply:
As how I'm doing the verification this is important.
For the dependency I'm taking the commit of the file (go.mod).
For the check code, I'm taking the commit of the folder.

This is in order to build only when necessary. By all means, if only one check changed but I'm taking latest commit from the repo, is like if dependencies has changed (even if not) and therefore all checks will be rebuild and pushed.
I think this is not what we want.

But I'm open to discuss via Slack.

Reply:
ok